dh/sidh/internal/templates/fp2.gotemp

```diff
@@ -79,10 +79,10 @@ func mul(dest, lhs, rhs *common.Fp2) {
 	// (b - a)*(c - d) = (b*c + a*d) - a*c - b*d
 	//
 	// so (a*d + b*c) = (b-a)*(c-d) + a*c + b*d.
-	mul{{ .FIELD}}(&ac, &lhs.A, &rhs.A)       // = a*c*R*R
-	mul{{ .FIELD}}(&bd, &lhs.B, &rhs.B)       // = b*d*R*R
 	sub{{ .FIELD}}(&bMinA, &lhs.B, &lhs.A)    // = (b-a)*R
 	sub{{ .FIELD}}(&cMinD, &rhs.A, &rhs.B)    // = (c-d)*R
+	mul{{ .FIELD}}(&ac, &lhs.A, &rhs.A)       // = a*c*R*R
+	mul{{ .FIELD}}(&bd, &lhs.B, &rhs.B)       // = b*d*R*R
```

Comment:
What's the reason for this reordering? As far as I can tell, this is to better line up with the order of operations in the comment above: `// so (a*d + b*c) = (b-a)*(c-d) + a*c + b*d.`

Reply:
The idea is to process multiplications together. Before there was an _interruption_ for processing subtractions.